eth2util/deposit/deposit.go

```diff
@@ -66,101 +60,87 @@ func GetMessageRoot(pubkey eth2p0.BLSPubKey, withdrawalAddr string) (eth2p0.Root
-		return eth2p0.Root{}, errors.Wrap(err, "withdrawal credentials")
-	}
-
-	dd := eth2p0.DepositData{
-		PublicKey:             pubkey,
-		WithdrawalCredentials: creds[:],
-		Amount:                eth2p0.Gwei(validatorAmt),
-		Signature:             sig,
-	}
-
-	hashRoot, err := dd.HashTreeRoot()
-	if err != nil {
-		return eth2p0.Root{}, errors.Wrap(err, "deposit data hash root")
-	}
-
-	return hashRoot, nil
-}
-
-// MarshalDepositData returns the json serialized DepositData.
-func MarshalDepositData(pubkey eth2p0.BLSPubKey, withdrawalAddr string, sig eth2p0.BLSSignature, network string) ([]byte, error) {
+// MarshalDepositData returns a json serialized deposit data.
+func MarshalDepositData(pubkey eth2p0.BLSPubKey, withdrawalAddr string, network string, msgSig eth2p0.BLSSignature) ([]byte, error) {
 	forkVersion := networkToForkVersion(network)
 
 	creds, err := withdrawalCredsFromAddr(withdrawalAddr)
 	if err != nil {
-		return nil, errors.Wrap(err, "withdrawal credentials")
+		return nil, err
 	}
 
 	// calculate depositMessage root
-	dmHashRoot, err := GetMessageRoot(pubkey, withdrawalAddr)
+	msgRoot, err := getMessageRoot(pubkey, withdrawalAddr)
 	if err != nil {
-		return nil, errors.Wrap(err, "deposit message root")
+		return nil, err
 	}
 
-	// calculate depositData root
-	ddHashRoot, err := GetDataRoot(pubkey, withdrawalAddr, sig)
+	// TODO(corver): Verify signature matches msgRoot.
+
+	dd := eth2p0.DepositData{
+		PublicKey:             pubkey,
+		WithdrawalCredentials: creds[:],
+		Amount:                validatorAmt,
+		Signature:             msgSig,
+	}
+	dataRoot, err := dd.HashTreeRoot()
 	if err != nil {
-		return nil, errors.Wrap(err, "deposit data root")
+		return nil, errors.Wrap(err, "deposit data hash root")
 	}
 
 	bytes, err := json.MarshalIndent(&depositDataJSON{
 		PubKey:                fmt.Sprintf("%x", pubkey),
 		WithdrawalCredentials: fmt.Sprintf("%x", creds),
 		Amount:                uint64(validatorAmt),
-		Signature:             fmt.Sprintf("%x", sig),
-		DepositMessageRoot:    fmt.Sprintf("%x", dmHashRoot),
-		DepositDataRoot:       fmt.Sprintf("%x", ddHashRoot),
+		Signature:             fmt.Sprintf("%x", msgSig),
+		DepositMessageRoot:    fmt.Sprintf("%x", msgRoot),
+		DepositDataRoot:       fmt.Sprintf("%x", dataRoot),
 		ForkVersion:           fmt.Sprintf("%x", forkVersion),
 		NetworkName:           network,
 	}, "", " ")
 	if err != nil {
-		return nil, errors.Wrap(err, "marshal deposit data json")
+		return nil, errors.Wrap(err, "marshal deposit data")
 	}
 
 	return bytes, nil
 }
 
-// GetDomain returns the Signature Domain.
-func GetDomain(forkVersion eth2p0.Version, domainType eth2p0.DomainType) (eth2p0.Domain, error) {
-	// For deposit, genesisValidatorsRoot is a fixed value.
-	var genesisValidatorsRoot eth2p0.Root
-	copy(genesisValidatorsRoot[:], zeroBytes32)
-
+// getDepositDomain returns the deposit signature domain.
+func getDepositDomain(forkVersion eth2p0.Version) (eth2p0.Domain, error) {
```

Comment:
more specific function signature, since only used in a single specific way

Reply:
makes sense👍